docs/content/sips/007-deployment-auth.md

```diff
@@ -0,0 +1,158 @@
+
+> QUESTION: Should we copy the user code to the clipboard?
+> * Pro: it's convenient
+> * Con: the user may already have stuff on their clipboard
+> * Con: again, documentation needs to consider differences across OSes
+
+(For both of these questions, is there prior art to which we can refer?)
+
+### Verification polling
+
+The device auth flow involves the client polling the server using the device code as the auth token. Each poll can have three outcomes:
+
+* 200 and the response contains a token: user has authorised this device
+* 200 and the response doesn't contain a token: user has not yet authorised this device
+* Server error
+* Network error
+
+We propose to poll for two minutes. If the user has not entered the code by then, `spin login` fails and exits.
+
+Errors do _not_ cause polling to terminate. We do not want a transient network error or server hiccup to block login. That said, if we can identify server responses that are necessarily fatal, we can in future terminate early on those.
+
+> QUESTION: Are there any accessibility considerations around the timeout? For example, should we provide a way to override it (e.g. longer timeout, poll only when the user is ready) for users who might not be able to execute the process within two minutes? Is there prior art from other applications we can refer to here?
+
+### Token storage
+
+When the user logs into a server, we need to store:
+
+* The server URL
+* Any server connection options, such as the `--insecure` flag
+* The token
+
+A simple solution is to store these in a file under a non-roaming directory. This is not fantastically secure for tokens, but may be enough for the preview.
+
+> QUESTION: What priority should we assign to storing tokens securely?  (E.g. DPAPI on Windows, not sure of the equivalent across other OSes.)
+
+> QUESTION: Should we provide a way for the user to control where things are stored, opening up the possibility of multiple configurations? (This also provides an approach to the default/override URL question on deploy.)
+
+### Programmatic control
+
+For environments like the VS Code extension, the standard interactive flow is hard to manage.  We therefore propose an alternative flow, controlled by flags that are hidden from interactive users, e.g.
+
+```
+$ spin login --url foo --get-device-code
+{ "verificationUrl" : "...", "deviceCode" : "...", "userCode": " ..." }
+
+$ spin login --url foo --check-device-code bar
+{ "status" : "waiting" }
+
+# user enters code into browser
+
+$ spin login --url foo --check-device-code bar
+{ "status" : "authorized", "token" : "..." }
+```
+
+> QUESTION: Should we present the token to the caller for it to store wherever it finds convenient, or store it ourselves in the same way as interactive login?  If the former, we will need a way for a caller to pass it to `spin deploy`, either via a new flag or by overloading `--hippo-password`.
```

Comment:
If we allow the credential store path to be overridden as above, the caller can point it at a temp location and read out from there.